compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -378,6 +381,30 @@ object Scanners {
         next.token = EMPTY
       }
 
+      def insertNL(nl: Token): Unit = {
+        next.copyFrom(this)
+        //  todo: make offset line-end of previous line?
+        offset = if (lineStartOffset <= offset) lineStartOffset else lastLineStartOffset
+        token = nl
+      }
+
+
+      /** A leading symbolic or backquoted identifier is treated as an infix operator
+       *  if it is followed by at least one ' ' and a token on the same line
+       *  that can start an expression.
+       */
+      def isLeadingInfixOperator =
+        allowLeadingInfixOperators &&
+        (token == BACKQUOTED_IDENT ||
+         token == IDENTIFIER && isOperatorPart(name(name.length - 1))) &&
+        (ch == ' ') && {
+          val lookahead = lookaheadScanner
+          lookahead.allowLeadingInfixOperators = false
+            // force a NEWLINE a after current token if it is on its own line
+          lookahead.nextToken()
+          canStartExpressionTokens.contains(lookahead.token)
+        }
```

Comment:
If we remove the field `allowLeadingInfixOperators`,  all the tests pass. However, the field makes a semantic difference for the following program:

```Scala
@main def Test = {
  val a = 5
  val x = 1
    + //
    `a` * 6

  assert(x == 1)
}
```

Maybe add the code above as a run test.